packages/tools/client-debugger/client-debugger-view/src/components/client-data-views/ContainerHistoryRecord.tsx

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { DefaultPalette, IStackItemStyles, Icon, Stack, StackItem } from "@fluentui/react";
+import React from "react";
+
+import {
+	ConnectionStateChangeLogEntry,
+	ContainerStateChangeKind,
+} from "@fluid-tools/client-debugger";
+
+/**
+ * Input props describing a Fluid {@link IFluidClientDebugger.getContainerConnectionLog}.
+ */
+export interface ContainerHistoryRecordViewProps {
+	/**
+	 * The status history of container to display.
+	 */
+	containerHistory: readonly ConnectionStateChangeLogEntry[];
+}
+
+/**
+ * Displays basic information about the provided {@link ContainerHistoryProps.containerHistory}.
+ */
+export function ContainerHistoryRecordView(
+	props: ContainerHistoryRecordViewProps,
+): React.ReactElement {
+	const { containerHistory } = props;
+	const nowTimeStamp = new Date();
+
+	// Reverse history such that newest events are displayed first
+	const reversedHistoryLog = [...containerHistory].reverse();
```

Comment:
small nit/question: I am unsure of the internals of `reverse()` and if it adds to operating complexity. Would be simpler to just traverse in reverse order?

Reply:
Good point. Yea it has O(n) complexity, I can just read from bottom.